### PR TITLE
direct_response: use data source

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Description: switch to using a data source for a more usable API
Risk Level: low (just recently merged the API)
Testing: unit
Docs Changes: none
Release Notes:

cc @mergeconflict

### CODE REVIEW

File: api/envoy/config/filter/network/direct_response/v2/config.proto [hunk 1]
```diff
@@ -15,6 +17,6 @@ option (udpa.annotations.file_migrate).move_to_package =
 // [#extension: envoy.filters.network.direct_response]
 
 message Config {
-  // Response data as bytes.
-  bytes response = 1;
+  // Response data as a data source.
+  api.v2.core.DataSource response = 1;
```
Comment (htuch) [inline api/envoy/config/filter/network/direct_response/v2/config.proto]: This is a breaking API change. You will need to add a new field for data source.

Comment (mattklein123) [reply api/envoy/config/filter/network/direct_response/v2/config.proto]: This was added in the last few days. No one is using it yet so IMO it's fine to change.

**Comment by kyessenov** [comment]
Weird coverage error once again:
```
All tests in the same test suite must use the same test fixture
class, so mixing TEST_F and TEST in the same test suite is
illegal.  In test suite UtilityTest,
test EnsureCreateSingletonsActuallyReturnsTheSameInstances is defined using TEST_F but
test ComputeHashedVersion is defined using TEST.  You probably
want to change the TEST to TEST_F or move it to another test
case.
```

**Comment by mattklein123** [comment]
Please merge master and it will fix that problem

**Comment by kyessenov** [comment]
Merged, still failing :( This time it looks legitimate, but I don't think I added any extra code paths.

**Comment by mattklein123** [comment]
@kyessenov hmm, dunno. I think the coverage build is pretty messed up right now and I wonder if the retries is causing strange results. Maybe we should remove that. cc @lizan @htuch 

/wait

**Comment by kyessenov** [comment]
PTAL. Coverage test de-flaked.

**Pull request merged**